packages/lib/types.ts

```diff
@@ -230,29 +230,132 @@ export const DefaultRiskScore = RiskScoreSchema.parse({
 })
 
 export const TokenMetaSchema = z.object({
-  type: z.string(),
-  category: z.string(),
-  description: z.string(),
-  displayName: z.string(),
-  displaySymbol: z.string(),
-  icon: z.string()
+  // Legacy fields
+  type: z.string().optional(),
+  icon: z.string().optional(),
+  // Core fields from yCMS
+  address: AddressSchema.optional(),
+  chainId: z.number().int().positive().optional(),
+  name: z.string().optional(),
+  symbol: z.string().optional(),
+  decimals: z.number().int().min(0).max(255).optional(),
+  // Display fields
+  displayName: z.string().optional(),
+  displaySymbol: z.string().optional(),
+  description: z.string().optional(),
+  category: z.string().optional(),
 })
 
 export type TokenMeta = z.infer<typeof TokenMetaSchema>
 
 export const VaultMetaSchema = z.object({
-  displayName: z.string(),
-  displaySymbol: z.string(),
-  description: z.string(),
-  protocols: z.string().array().nullish()
+  chainId: z.number(),
+  address: AddressSchema,
+  name: z.string(),
+  registry: AddressSchema.optional(),
+  type: z.enum(['Yearn Vault', 'Experimental Yearn Vault', 'Automated Yearn Vault', 'Single Strategy', 'None']),
```

Comment:
for all the enum fields, lets just use z.string in the kong\client schema. otherwise adding an enum value in cms requires updating kong too which seems unnecessary.